File: samples/sample-pdk-pipeline-java/src/main/java/software/aws/ApplicationStage.java
package software.aws;

import org.jetbrains.annotations.Nullable;
import software.amazon.awscdk.Stage;
import software.amazon.awscdk.StageProps;
import software.constructs.Construct;

public class ApplicationStage extends Stage {

    protected ApplicationStage(Construct scope, String id, @Nullable StageProps props) {
        super(scope, id, props);

        new ApplicationStack(this, "MyAplication", null);
    }
}

File: packages/pipeline/samples/typescript/src/application-stage.ts
/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { Stage, StageProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { ApplicationStack } from './application-stack';

export class ApplicationStage extends Stage {
  constructor(scope: Construct, id: string, props?: StageProps) {
    super(scope, id, props);

    new ApplicationStack(this, 'MyApplication');
  }
}